src/resolve/resolved-dns-stream.c

```diff
@@ -183,6 +183,36 @@ static int dns_stream_identify(DnsStream *s) {
         return 0;
 }
 
+static ssize_t dns_stream_writev(DnsStream *s, struct iovec *iov, int iovcnt) {
+        ssize_t r;
+
+        assert(s);
+        assert(iov);
+
+        if (s->tfo_salen) {
```

Comment:
we usually avoid relying on C's downgrade-to-bool feature for numeric values that aren't actually conceptually booleans (except for pointer checks). Since this is a byte length, hence please use "if (s->tfo_salen > 0)"